docs/DeveloperGuide.md

```diff
@@ -2,12 +2,361 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+[Original Source](https://github.com/nus-cs2113-AY2425S1/tp)
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Overall Class Diagram
+![overallClass.drawio.png](./Diagrams/Overall/overallClass.drawio.png)
```

Comment:
I agree that an architecture diagram might be more effective in displaying a clearer high-level overview of how the system components interact with each other. The detailed information in the class diagram, while useful, may be more than necessary for showing the overall structure and interactions of the system’s major components.